### PR TITLE
Version bump to chocolatey v2.1.0

### LINKED ISSUE #157: Chocolatey version 2.0.0
Would be nice to be able to use the v2.0.0 of Chocolatey with this GH action.

Goal is to use GitHub packages to host Chocolatey packages since v2.0.0 of Chocolatey now supports NuGet v3 feeds.

Probably just need to change the version number here: https://github.com/crazy-max/ghaction-chocolatey/blob/84f876ce8c3cb6f320b1ca03f48a47c10d9e3a15/image/Dockerfile#L3C1-L4C1

### PR DESCRIPTION
**Pull request by regg00**
Fix #157 

@crazy-max The official Choco repo is currently not building because of a problem with GitVersion so I had to create a patch.
I kept getting this error message in the build logs: `System.InvalidOperationException: Gitversion could not determine which branch to treat as the development branch (default is 'develop') nor release-able branch (default is 'main' or 'master'), either locally or remotely. Ensure the local clone and checkout match the requirements or considering using 'GitVersion Dynamic Repositories'`

I'm pretty sure someone smarter than me can create a better fix, but what I did is working.

### CODE REVIEW

**Comment by regg00** [comment]
See this commit for more info: https://github.com/chocolatey/choco/commit/2bbe5d711ad1499fef0cb24b70ef4b87ff7fd505

**Comment by regg00** [comment]
I opened an issue on the main choco repo https://github.com/chocolatey/choco/issues/3243
Seems that cloning a specific branch causes the issue (ex: `git clone --branch 2.1.0 https://github.com/chocolatey/choco.git`)
If I clone the main repo with `git clone https://github.com/chocolatey/choco.git` it builds without the patch to the GitVersion file.

**Comment by regg00** [comment]
Based on this recommendation here https://github.com/chocolatey/choco/issues/3243#issuecomment-1615604532 I modified the Dockerfile to download pre-built binaries from GitHub releases.

I think it's a more robust and simple solution.
@crazy-max Let me know what you think.

**Comment by crazy-max** [comment]
@regg00 Thanks for working on this!

> Based on this recommendation here [chocolatey/choco#3243 (comment)](https://github.com/chocolatey/choco/issues/3243#issuecomment-1615604532) I modified the Dockerfile to download pre-built binaries from GitHub releases.

Ah they have an official Docker image now, that's handy. Moving forward I think we could just add a `version` input so it would download chocolatey directly from https://github.com/chocolatey/choco/releases so on Windows runners it would install the desired version (if provided) from the MSI or we could just run the official image that seems to support windows too, otherwise it uses the default one installed on the runner. For Linux, if version is provided it takes the tarball from GH release or the image. I think we could just use the official image as version to be fair so by default it would be `latest` and you can put whatever tag you want to use. I will make some changes related to this and keep you posted.

**Comment by crazy-max** [comment]
Completely forgot we already had the `image` input. Can you try with:

```yaml
      -
        name: Choco help
        uses: crazy-max/ghaction-chocolatey@v2
        with:
          image: chocolatey/choco:v2.1.0
          args: -h
```

**Comment by crazy-max** [comment]
Oh boy looking at the windows image we can't afford using it looking at its size:

image

**Comment by crazy-max** [comment]
Even the linux is quite huge compared to mine: https://hub.docker.com/r/crazymax/ghaction-chocolatey/tags?page=1&ordering=last_updated

image

There should be room for improvements with the official image

**Comment by crazy-max** [comment]
Will take your PR in the meantime but moving forward I think I will also provide a slim Windows image and create a manifest list for multi-platform support.

File: image/Dockerfile [hunk 1]
```diff
@@ -1,40 +1,25 @@
...
-  && apt-get install -y -t buster-backports git
 
-ENV ChocolateyInstall="/opt/chocolatey"
-COPY --from=src /src /usr/local/src/choco
-WORKDIR /usr/local/src/choco
-RUN ./build.sh --verbosity=diagnostic
 
 FROM alpine:${ALPINE_VERSION}
-COPY --from=builder /usr/local/src/choco/code_drop/temp/_PublishedApps/choco /opt/chocolatey
+ARG CHOCOLATEY_VERSION
```
Comment (crazy-max) [inline image/Dockerfile:9]: Should be placed right before `RUN wget` otherwise cache will be evicted for `apk` run.

File: image/Dockerfile [hunk 2]
```diff
@@ -1,40 +1,25 @@
...
+WORKDIR /opt/chocolatey
+
+RUN apk add --no-cache wget
```
Comment (crazy-max) [inline image/Dockerfile:12]: Would prefer curl:

```suggestion
RUN apk add --no-cache curl
```

File: image/Dockerfile [hunk 3]
```diff
@@ -1,40 +1,25 @@
...
 RUN apk --update --no-cache --repository https://dl-cdn.alpinelinux.org/alpine/edge/testing add mono-dev \
   && apk --update --no-cache add -t build-dependencies ca-certificates \
   && cert-sync /etc/ssl/certs/ca-certificates.crt \
   && ln -sf /opt /opt/chocolatey/opt \
   && mkdir -p /opt/chocolatey/lib \
   && apk del build-dependencies \
   && rm -rf /var/cache/apk/*
+
+RUN wget https://github.com/chocolatey/choco/releases/download/${CHOCOLATEY_VERSION}/chocolatey.v${CHOCOLATEY_VERSION}.tar.gz
+RUN tar -zxvf chocolatey.v${CHOCOLATEY_VERSION}.tar.gz
```
Comment (crazy-max) [inline image/Dockerfile:22]: Oneline cmd:

```suggestion
RUN curl -sSL "https://github.com/chocolatey/choco/releases/download/${CHOCOLATEY_VERSION}/chocolatey.v${CHOCOLATEY_VERSION}.tar.gz" | tar xz
```

**Review by crazy-max — CHANGES_REQUESTED** [review]

**Comment by regg00** [comment]
Thanks for the suggestions. 
I edited the code accordingly.

**Pull request closed without merging**